// lunr-languages/test/testdata/ru.js
module.exports = {
    fields: [
        {
            name: 'title',
            config: { boost: 10 }
        }, {
            name: 'body'
        }
    ],
    documents: [
        {
            "title": "Россия",
            "body": "Росси́я (от греч. Ρωσία — Русь[15]), Росси́йская Федера́ция[К 9] (на практике используется и аббревиатура РФ[К 10]) — государство в Восточной Европе и Северной Азии. Население — 146 804 372[4] чел. (2017). Территория России, определяемая её Конституцией, составляет 17 125 191[3][К 4] км². Занимает первое место в мире по территории, шестое — по объёму ВВП по ППС и девятое — по численности населения.",
            "id": 1
        }, {
            "title": "Происхождение названия",
            "body": "Первое письменное упоминание термина «Росия» (греч. Ρωσία) датировано серединой Х века. Он встречается в сочинениях византийского императора Константина Багрянородного «О церемониях» и «Об управлении империей» как греческое название Руси[53]. В кириллической записи слово «Росия» (Рѡсїѧ) было впервые употреблено 24 апреля 1387 года[54] в титуле митрополита Киприана, собственноручно подписавшегося как «митрополит Кыевскый и всея Росия». В XV—XVI веках эллинизированное название «Росия» закрепилось за русскими землями, которые были объединены в единое государство под началом Московского княжества. Официальный статус оно приобрело после венчания Ивана IV на царство в 1547 году, когда государство стало называться Российским царством[55]. Современное написание слова — с двумя буквами «С» — появилось с середины XVII века и окончательно закрепилось при Петре I. павловна",
            "id": 2
        }
    ],
    tests: [
        {
            what: "find the word %w",
            search: "Росси́йская",
            found: 1
        }, {
            what: "find the word %w",
            search: "упоминание",
            found: 1
        }, {
            what: "never find a word that does not exist, like %w",
            search: "inexistent",
            found: 0
        }, {
            what: "find a correctly stemmed word %w",
            search: "павловне",
            found: 1
        }
    ]
}
